src/Client.php

```diff
@@ -2,89 +2,58 @@
 
 namespace MeiliSearch;
 
-class Client extends HTTPRequest
+use MeiliSearch\Delegates\handlesIndex;
+use MeiliSearch\Delegates\handlesSystem;
+use MeiliSearch\Endpoints\Health;
+use MeiliSearch\Endpoints\Indexes;
+use MeiliSearch\Endpoints\Keys;
+use MeiliSearch\Endpoints\Stats;
+use MeiliSearch\Endpoints\SysInfo;
+use MeiliSearch\Endpoints\Version;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+
+class Client
 {
-    // Indexes
-
-    public function getAllIndexes()
-    {
-        return $this->httpGet('/indexes');
-    }
-
-    public function showIndex($uid)
-    {
-        return $this->indexInstance($uid)->show();
-    }
-
-    public function deleteIndex($uid)
-    {
-        return $this->indexInstance($uid)->delete();
-    }
-
-    public function deleteAllIndexes()
-    {
-        foreach ($this->getAllIndexes() as $index) {
-            $this->deleteIndex($index['uid']);
-        }
-    }
-
-    public function getIndex($uid)
-    {
-        return $this->indexInstance($uid);
-    }
-
-    public function createIndex($index_uid, $options = [])
-    {
-        $body = array_merge(
-            $options,
-            ['uid' => $index_uid]
-        );
-        $response = $this->httpPost('/indexes', $body);
-        $uid = $response['uid'];
-
-        return $this->indexInstance($uid);
-    }
-
-    // Health
-
-    public function health()
-    {
-        return $this->httpGet('/health');
-    }
-
-    // Stats
-
-    public function version()
-    {
-        return $this->httpGet('/version');
-    }
-
-    public function sysInfo()
-    {
-        return $this->httpGet('/sys-info');
-    }
-
-    public function prettySysInfo()
-    {
-        return $this->httpGet('/sys-info/pretty');
-    }
-
-    public function stats()
-    {
-        return $this->httpGet('/stats');
-    }
-
-    // Keys
-
-    public function getKeys()
-    {
-        return $this->httpGet('/keys');
-    }
-
-    // Private methods
-
-    private function indexInstance($uid)
-    {
-        return new Index($uid, $this->base_url, $this->api_key);
+    use handlesIndex;
+    use handlesSystem;
+
+    private $http;
+    /**
+     * @var Indexes
+     */
+    private $index;
+
+    /**
+     * @var Health
+     */
+    private $health;
+    /**
+     * @var Version
+     */
+    private $version;
+    /**
+     * @var SysInfo
+     */
+    private $sysInfo;
+    /**
+     * @var Keys
+     */
+    private $keys;
+    /**
+     * @var Stats
+     */
+    private $stats;
+
+    public function __construct($url, $apiKey = null, ClientInterface $httpClient = null, RequestFactoryInterface $requestFactory = null, StreamFactoryInterface $streamFactory = null)
```

Comment:
Maybe you can add types for each arguments, especially the first ones which are simple strings 🙂

Reply:
done